deploy.sh

```
@@ -0,0 +1,179 @@
+#! /usr/bin/env nix-shell
+#! nix-shell -p jq -i bash
+
+set -euo pipefail
+ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )/.."
+
+SSHOPTS=(-o 'StrictHostKeyChecking=no' -o 'UserKnownHostsFile=/dev/null')
+export NIX_SSHOPTS="${SSHOPTS[*]}"
+
+# For first deployment to bare server
+if [[ ${1:-} == '--prime' ]]; then
+    shift
+
+    BARE_SERVER=1
+    TARGET_USER=root
+    LOCAL_KEY="${LOCAL_KEY-$(readlink -f "nix.key")}"
+
+    if [[ ! -r $LOCAL_KEY ]]; then
+        echo "Couldn't find nix.key. Required to locally sign closures. Aborting."
+        exit 1
+    fi
+fi
+
+
+##
+# Push the closure to the server using `nix copy`. Sign closure first if
+# deploying from workstation.
+function push_closure() {
+    if [[ -z ${HOST:-} ]]; then
+        echo 'Missing target host.'
+        exit 1
+    fi
+
+    if [[ -z ${USER:-} ]]; then
+        echo 'Missing user'
+        exit 1
+    fi
+
+    if [[ -z ${CLOSURE:-} ]]; then
+        echo 'Missing closure store path'
+        exit 1
+    fi
+
+    if [[ ! -d $CLOSURE ]]; then
+        nix build $CLOSURE
+    fi
+
+    ##
+    # First deployment is done from local workstation, sign keys explicitly.
+    if [[ -n ${LOCAL_KEY:-} ]]; then
+        nix sign-paths -r -k "$LOCAL_KEY" "$CLOSURE"
+    fi
+
+    nix copy --substitute-on-destination --to "ssh://$USER@$HOST" "$CLOSURE"
+}
+
+function activate_profile() {
+    if [[ -z ${HOST:-} ]]; then
+        echo 'Missing target host.'
+        exit 1
+    fi
+
+    if [[ -z ${USER:-} ]]; then
+        echo 'Missing user name'
+        exit 1
+    fi
+
+    if [[ ! -d ${CLOSURE:-} ]]; then
+        echo 'Missing or invalid closure store path'
+        exit 1
+    fi
+
+    if [[ -z ${PROFILE_PATH:-} ]]; then
+        echo 'Missing profile path'
+        exit 1
+    fi
+
+    ssh "${SSHOPTS[@]}" "$USER@$HOST" "export PROFILE=$PROFILE_PATH; $SUDO nix-env -p $PROFILE_PATH --set $CLOSURE; eval \"$SUDO $ACTIVATE\""
```

Should be `&&` instead of `;` to fail early